### ISSUE TITLE
Aer cannot simulate circuit using only few qubits if width is large

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-aer#218]
### Informations

- **Qiskit Aer version**: master (0.3)
- **Python version**:
- **Operating system**: osx

### What is the current behavior?
```python

circ = QuantumCircuit(50, 1)
circ.h(45)
for _ in range(100):
     circ.t(45)
circ.measure(45, 0)
job = execute(circ, Aer.get_backend('qasm_simulator'))
```

gives:

```python

AerError: 'ERROR: a circuit requires more memory than max_memory_mb.'
```

Also:

```python

import qiskit
from qiskit.quantum_info.random import random_unitary
from qiskit import Aer

U = random_unitary(4)
qr = QuantumRegister(50)
cr = ClassicalRegister(2)
qc = QuantumCircuit(qr, cr)
qc.append(U,[qr[48],qr[49]], [])
qc.measure(qr[48], cr[0])
qc.measure(qr[49], cr[1])
backend = Aer.get_backend('qasm_simulator')
job = qiskit.execute(qc, backend=backend)
job.result()
```

gives:

```python

AerError: 'ERROR: QasmController: Circuit cannot be run on any available backend. max_memory_mb=8192mb'
```

**Comment by ajavadia** [comment]
This was working for me, but not for @nonhermitian. Our versions seemed identical, so not sure what the issue is. If others can try this and report the result with version number that'd be great.

**Comment by chriseclectic** [comment]
I think the issue here is that the memory checks are happening before the qubit truncation pass. Currently all circuit optimiztion passes happen at a single step just before execution (but after all the memory checks for automatic configuration of paralellization). We need to change how these passes are implemented so that some, such as this truncation, can be applied before the memory checks.

**Comment by chriseclectic** [comment]
Closed by #296
